src/librustc_data_structures/transitive_relation.rs

```diff
@@ -100,12 +99,12 @@ impl<T: Clone + Debug + Eq + Hash> TransitiveRelation<T> {
         for edge in &self.edges {
             let r = f(&self.elements[edge.source.0]).and_then(|source| {
                 f(&self.elements[edge.target.0]).and_then(|target| {
-                    Some(result.add(source, target))
+                    result.add(source, target);
+                    Some(())
                 })
             });
-            if r.is_none() {
-                return None;
-            }
+
+            r?;
```

Comment:
remove the local variable `r` entirely and add the question mark to the function call